pkg/sdk/functions_gen_test.go

```diff
@@ -0,0 +1,528 @@
+	})
+
+	t.Run("alter: set log level", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Set = &FunctionSet{
+			LogLevel: String("DEBUG"),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET LOG_LEVEL = 'DEBUG'`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: set trace level", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Set = &FunctionSet{
+			TraceLevel: String("DEBUG"),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET TRACE_LEVEL = 'DEBUG'`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: set comment", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Set = &FunctionSet{
+			Comment: String("comment"),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET COMMENT = 'comment'`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: set secure", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Set = &FunctionSet{
+			Secure: Bool(true),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET SECURE`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: unset secure", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Unset = &FunctionUnset{
+			Secure: Bool(true),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) UNSET SECURE`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: unset comment", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Unset = &FunctionUnset{
+			Comment: Bool(true),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) UNSET COMMENT`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: set tags", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.SetTags = []TagAssociation{
+			{
+				Name:  NewAccountObjectIdentifier("tag1"),
+				Value: "value1",
+			},
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) SET TAG "tag1" = 'value1'`, id.FullyQualifiedName())
+	})
+
+	t.Run("alter: unset tags", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.UnsetTags = []ObjectIdentifier{
+			NewAccountObjectIdentifier("tag1"),
+			NewAccountObjectIdentifier("tag2"),
+		}
+		assertOptsValidAndSQLEquals(t, opts, `ALTER FUNCTION IF EXISTS %s (VARCHAR, NUMBER) UNSET TAG "tag1", "tag2"`, id.FullyQualifiedName())
+	})
+}
+
+func TestFunctions_Show(t *testing.T) {
+	defaultOpts := func() *ShowFunctionOptions {
+		return &ShowFunctionOptions{}
+	}
+
+	t.Run("validation: nil options", func(t *testing.T) {
+		opts := (*ShowFunctionOptions)(nil)
+		assertOptsInvalidJoinedErrors(t, opts, ErrNilOptions)
+	})
+
+	t.Run("validation: empty like", func(t *testing.T) {
+		opts := defaultOpts()
+		opts.Like = &Like{}
+		assertOptsInvalidJoinedErrors(t, opts, ErrPatternRequiredForLikeKeyword)
+	})
+
+	t.Run("show with empty options", func(t *testing.T) {
+		opts := defaultOpts()
+		assertOptsValidAndSQLEquals(t, opts, `SHOW USER FUNCTIONS`)
```

Comment:
oh, ok. So you meant show user functions. There was some misunderstanding from my side in the def.go file, because you have there in the documentation pointing to /show-functions endpoint, but that way I've  found that there are 2 show functions 
https://docs.snowflake.com/en/sql-reference/sql/show-user-functions
and
https://docs.snowflake.com/en/sql-reference/sql/show-functions